org.lflang/src/org/lflang/validation/LFValidator.java

```diff
@@ -1394,31 +1418,42 @@ public void checkStateResetWithoutInitialValue(StateVar state) {
     //////////////////////////////////////////////////////////////
     //// Public methods.
 
-    /** 
-     * Return the error reporter for this validator. 
+    /**
+     * Return the error reporter for this validator.
      */
     public ValidatorErrorReporter getErrorReporter() {
         return this.errorReporter;
     }
-    
+
     /**
      * Implementation required by xtext to report validation errors.
      */
     @Override
     public ValidationMessageAcceptor getMessageAcceptor() {
         return messageAcceptor == null ? this : messageAcceptor;
     }
-    
+
     /**
      * Return a list of error messages for the target declaration.
      */
     public List<String> getTargetPropertyErrors() {
         return this.targetPropertyErrors;
     }
 
+    /**
+     * Generate an error message for an AST node.
+     * Overriding the parent method here to make the method public,
+     * so that it can be called from the AttributeSpec class.
+     */
+    @Override
+    public void error(java.lang.String message, 
```

Comment:
Technically you can also leave it protected. Protected visibility is subclasses + current package, so it would be visible in AttributeSpec, which is in this package. This is a detail though